azurelinuxagent/common/cgroup.py

```diff
@@ -161,24 +145,39 @@ def _get_cpu_ticks(self, allow_no_such_file_or_directory_error=False):
         returns 0; this is useful when the function can be called before the cgroup has been created.
         """
         try:
-            cpu_stat = self._get_file_contents('cpuacct.stat')
+            cpuacct_stat = self._get_file_contents('cpuacct.stat')
         except Exception as e:
             if not isinstance(e, (IOError, OSError)) or e.errno != errno.ENOENT:  # pylint: disable=E1101
                 raise CGroupsException("Failed to read cpuacct.stat: {0}".format(ustr(e)))
             if not allow_no_such_file_or_directory_error:
                 raise e
-            cpu_stat = None
+            cpuacct_stat = None
 
         cpu_ticks = 0
 
-        if cpu_stat is not None:
-            match = re_user_system_times.match(cpu_stat)
+        if cpuacct_stat is not None:
+            match = re_user_system_times.match(cpuacct_stat)
             if not match:
-                raise CGroupsException("The contents of {0} are invalid: {1}".format(self._get_cgroup_file('cpuacct.stat'), cpu_stat))
+                raise CGroupsException("The contents of {0} are invalid: {1}".format(self._get_cgroup_file('cpuacct.stat'), cpuacct_stat))
             cpu_ticks = int(match.groups()[0]) + int(match.groups()[1])
 
         return cpu_ticks
 
+    def _get_throttled_time(self):
+        try:
+            with open(os.path.join(self.path, 'cpu.stat')) as cpu_stat:
+                for line in cpu_stat:
+                    match = re.match(r'throttled_time\s+(\d+)', line)
+                    if match is not None:
+                        return int(match.groups()[0])
+                raise Exception("Cannot find throttled_time")
+        except (IOError, OSError) as e:
+            if e.errno == errno.ENOENT:
+                return 0
```

Comment:
I see that you're returning 0 if FileNotFound. Wouldn't this be misconstrued as the agent was not throttled at all?

Reply:
the file wouldn't exist if the cpu controller wasn't active, and then there would not be any throttling

Reply:
also, this won't happen on the agent, but when an extension completes the cgroup and associated files go away after a short time. we should probably detect that case and just don't send telemetry (will require some kind of communication within the extension handler and the monitor thread, though)

Reply:
Ahh makes sense

Reply:
maybe we can just let the not found error bubble up to the monitor thread and check there if the cgroup still exists and cleanup there. we already have some cleanup logic that i wanted to hook up to cgroup notifications... i'll make a note, all of that would be for extensions